src/control/ControlUtils.cpp

```diff
@@ -101,14 +102,35 @@ Vector2 ControlUtils::projectPositionToOutsideDefenseArea(const world::Field &fi
 
 /// Calculates the chip force
 double ControlUtils::determineChipForce(const double distance, stp::KickChipType desiredBallSpeedType) noexcept {
-    // TODO: For now, we just return the max_chip_power, but this needs tuning
-    return stp::control_constants::MAX_CHIP_POWER;
+    // TODO: Needs further tuning
+    constexpr double TARGET_FACTOR{0.5};
+    constexpr double PASS_FACTOR{0.745};
+
+    if (desiredBallSpeedType == stp::MAX) return stp::control_constants::MAX_KICK_POWER;
+
+    double limitingFactor{};
+    // Pick the right limiting factor based on ballSpeedType and whether we use GRSIM or not
+    if (desiredBallSpeedType == stp::PASS) {
+        limitingFactor = PASS_FACTOR;
+    } else if (desiredBallSpeedType == stp::TARGET) {
+        limitingFactor = TARGET_FACTOR;
+    } else {
+        RTT_ERROR("No valid ballSpeedType, kick velocity set to 0")
+        return 0;
+    }
+
+    // Calculate the velocity based on this function with the previously set limitingFactor
+    auto velocity = distance * limitingFactor;
+
+    // Make sure velocity is always between MIN_KICK_POWER and MAX_KICK_POWER
+    return std::clamp(velocity, stp::control_constants::MIN_KICK_POWER, stp::control_constants::MAX_KICK_POWER);
```

Comment:
Shouldn't we make sure the calculation doesn't go outside of those bounds, rather than clamp it?

Reply:
I think the speed calculation in general could use a new algorithm, so I will just leave this for now.